docs/usage/schema.md

```diff
@@ -131,6 +131,11 @@ For versions of Python prior to 3.9, `typing_extensions.Annotated` can be used.
 Custom field types can customise the schema generated for them using the `__modify_schema__` class method;
 see [Custom Data Types](types.md#custom-data-types) for more details.
 
+You can also add any of the following arguments to the signature to use them in the implementation:
+
+* `field`: the field whose schema is customized. Type of object is `pydantic.fields.ModelField`.
+* `**kwargs`: if provided, the above argument will be provided in they `kwargs` dictionary
```

Comment:
```suggestion
`__modify_schema__` can also take a `field` argument which will have type `Optional[ModelField]`, pydantic will inspect the signature of `__modify_schema__` to determine
whether the `field` argument should be included.
```
Or something. Explaining about `**kwargs` is overkill here I think.

Might also be worth adding an example.

Reply:
Simplified the documentation as suggested, and added an example